ccvm_simulators/solvers/dl_solver.py

```diff
@@ -228,6 +276,84 @@ def tune(self, instances, post_processor=None, pump_rate_flag=True, g=0.05):
         #       future consideration
         self.is_tuned = True
 
+    def _optics_machine_energy(self, machine_parameters=None):
+        """The wrapper function of calculating the average energy consumption of the
+            solver, as if the solving process was to be performed on an optical DL-CCVM
+            machine.
+
+        Args:
+            machine_parameters (dict, optional): Parameters of the machine. Defaults to None.
+
+        Raises:
+            ValueError: when the given machine parameters are not valid.
+            ValueError: when the given dataframe does not contain the required columns.
+
+        Returns:
+            Callable: A callable function that takes in a dataframe and problem size and
+                returns the average energy consumption of the solver.
+        """
+
+        if machine_parameters is None:
+            machine_parameters = self._default_optics_machine_parameters
+        else:
+            self._is_valid_optics_machine_parameters(machine_parameters)
+
+        def _optics_machine_energy_callable(matching_df: DataFrame, problem_size: int):
+            """Calculate the average energy consumption of the solver simulating on a
+                DL-CCVM machine.
+
+            Args:
+                matching_df (DataFrame): The necessary data to calculate the average energy.
+                problem_size (int): The size of the problem.
+
+            Raises:
+                ValueError: when the given dataframe does not contain the required columns.
+
+            Returns:
+                float: The average energy consumption of the solver.
+            """
+            if "solve_time" not in matching_df.columns:
+                raise ValueError(
+                    "The given dataframe does not contain the column 'solve_time'"
+                )
+            self._validate_dataframe_columns(matching_df)
+
+            try:
+                pump = self.parameter_key[problem_size]["pump"]
+            except KeyError as e:
+                raise KeyError(
+                    f"Pump for the given instance size: {problem_size} is not defined."
+                )
+
+            T_clock = machine_parameters["laser_clock"]
+            P_opt = machine_parameters["laser_power"]
+            T_elec = machine_parameters["electronics_latency"]
+            P_mod = machine_parameters["modulators_power"]
+            P_sq = machine_parameters["squeezing_power"]
+            P_elec = machine_parameters["electronics_power"]
+            P_opa = machine_parameters["amplifiers_power"]
+            postprocessing_time = np.mean(matching_df["pp_time"].values)
+            iterations = matching_df["iterations"].values
```

Comment:
This is the line I've asked to change in the other PR to deal with the error in plotting ETS in `dl-ccvm`.
```suggestion
            iterations = np.mean(matching_df["iterations"].values)
```

Reply:
Right. I've updated both places for both `dl-ccvm` and `mf-ccvm`.